File: VB/DXSample/ViewModel/NavigationItemViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DXSample.Common
Imports System.Windows.Media.Imaging

Namespace DXSample.ViewModel
    Public Class NavigationItemViewModel

        Public Overridable Property Title() As String
        Public Overridable Property Image() As BitmapImage
        Public Overridable Property ModuleType() As ModuleType

        Protected Sub New()
            ViewInjectionManager.Default.RegisterNavigatedEventHandler(Me, Sub() ViewInjectionManager.Default.Navigate(Regions.Main, ModuleType))
        End Sub
        Public Shared Function Create(ByVal title As String, ByVal image As BitmapImage, ByVal type As ModuleType) As NavigationItemViewModel
            Dim t = ViewModelSource.Create(Function() New NavigationItemViewModel())
            t.Title = title
            t.Image = image
            t.ModuleType = type
            Return t
        End Function
    End Class
End Namespace

File: CS/DXSample/ViewModel/NavigationItemViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DXSample.Common;
using System.Windows.Media.Imaging;

namespace DXSample.ViewModel {
    public class NavigationItemViewModel {
        
        public virtual string Title { get; set; }
        public virtual BitmapImage Image { get; set; }
        public virtual ModuleType ModuleType { get; set; }

        protected NavigationItemViewModel() {
            ViewInjectionManager.Default.RegisterNavigatedEventHandler(this, () => {
                ViewInjectionManager.Default.Navigate(Regions.Main, ModuleType);
            });
        }
        public static NavigationItemViewModel Create(string title, BitmapImage image, ModuleType type) {
            var t = ViewModelSource.Create(() => new NavigationItemViewModel());
            t.Title = title;
            t.Image = image;
            t.ModuleType = type;
            return t;
        }
    }
}
